apps/wallet/src/engine/runners/tezos/index.ts

```diff
@@ -1,16 +1,98 @@
+import {} from '@taquito/tzip16';
+import type { TezosToken } from '@walless/core';
+import { Networks } from '@walless/core';
+import type { PublicKeyDocument, TokenDocument } from '@walless/store';
+import { selectors } from '@walless/store';
+import { getTokenQuote } from 'utils/api';
+import { addTokenToStorage, storage } from 'utils/storage';
+
 import type { EngineConfig, Runner } from '../../types';
 
+import { constructTezosTokenDocument, XTZ } from './token';
+import { convertTezosImageUriToUrl, getTezosContext } from './utils';
+
 export type TezosContext = {
-	connection: unknown;
+	connection: TezosToolkit;
+	tzktApi: string;
 };
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-export const createTezosRunner = (config: EngineConfig): Runner => {
+export const createTezosRunner = async (
+	config: EngineConfig,
+): Promise<Runner> => {
+	const { networkClusters } = config;
+	const cluster = networkClusters[Networks.tezos];
+	const { connection, tzktApi } = getTezosContext(config);
+
+	const keysResult = await storage.find<PublicKeyDocument>(selectors.tezosKeys);
+	const keys = keysResult.docs;
+
 	return {
-		start() {},
+		async start() {
+			const tokensPromises = keys.map(async (key) => {
+				const owner = key._id;
+				const tzBalance = await connection.tz.getBalance(owner);
+
+				const tokenBalances = await fetch(
+					`${tzktApi}/tokens/balances?account=${owner}`,
+				);
+				const tokenBalancesJson = await tokenBalances.json();
+
+				const tzDocument = constructTezosTokenDocument({
+					...XTZ,
+					owner: key._id,
+					cluster,
+					amount: tzBalance.toString(),
+					balance: parseFloat(tzBalance.toString()) / 10 ** XTZ.decimals,
+				} as TezosToken);
+
+				const tokenDocuments: TokenDocument<TezosToken>[] = [tzDocument];
+
+				tokenBalancesJson.forEach((tokenBalance) => {
+					const tokenDocument = constructTezosTokenDocument({
+						owner: key._id,
+						network: Networks.tezos,
+						cluster,
+						tokenId: tokenBalance.token.tokenId,
+						contract: tokenBalance.token.contract,
+						tokenType: tokenBalance.token.standard,
+						decimals: tokenBalance.token.metadata.decimals,
+						symbol: tokenBalance.token.metadata.symbol,
+						name: tokenBalance.token.metadata.name,
+						amount: tokenBalance.balance,
+						balance:
+							parseFloat(tokenBalance.balance) /
+							10 ** tokenBalance.token.metadata.decimals,
+						image: convertTezosImageUriToUrl(
+							tokenBalance.token.metadata.thumbnailUri,
+						),
+					} as TezosToken);
+
+					console.log(tokenDocument);
+
+					tokenDocuments.push(tokenDocument);
+				});
+
+				for (const tokenDocument of tokenDocuments) {
+					const responseQuotes = await getTokenQuote({
+						address: `${tokenDocument.contract?.address}`,
+						network: tokenDocument.network,
+					});
+					tokenDocument.quotes = responseQuotes?.quotes;
+
+					await addTokenToStorage<TokenDocument<TezosToken>>(tokenDocument);
+				}
```

Comment:
I could separate loading quote for each token promise to prevent UI blocking (UI updated only if all tokens loaded)